ios/browser/browser_state/brave_browser_state_manager_impl.cc

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/browser_state/brave_browser_state_manager_impl.h"
+
+#include "brave/ios/browser/url_sanitizer/url_sanitizer_service_factory.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+
+void BraveBrowserStateManagerImpl::DoFinalInitForServices(
+    ChromeBrowserState* browser_state) {
+  ChromeBrowserStateManagerImpl::DoFinalInitForServices(browser_state);
+  brave::URLSanitizerServiceFactory::GetServiceForState(browser_state);
```

Comment:
I'm not understanding what you're doing here. Why would you call this method and then do nothing with the result?

Reply:
Also noticing this pattern of `Factory::GetServiceForState` in several places that we should not be following here. It seems to be copied from wallet in both desktop and ios (cc @darkdh and @kylehickinson) where it should be `GetInstance()->GetServiceForContext` and `GetInstance()->GetServiceForState` respectively. Those factories have accessors for mojo and (for internal use only) raw pointers, but URLSanitizerService does not have those and should use the standard `GetForBrowserState` method

Reply:
@kylehickinson @cuba said you asked him to do this for early init, but that should be accomplished by overriding `BrowserStateKeyedServiceFactory::ServiceIsCreatedWithBrowserState` to return true which will initialize it here https://github.com/brave/brave-core/pull/19079/files#diff-0a7a471c5e606a0fc0597d565013dc0080574a16a8d6e004f3b3a570a963b1bcR27

Reply:
I don't know what to do here. The file name "url_sanitizer_service_factory.h" is already taken in the same folder, this is a objc file which represent the url_sanitizer_service_factory class but I can't expose this class directly to iOS hence this wrapper. I really don't think we want "Wrapper" at the end of our objc classes in iOS.

Reply:
@bridiver So the whole thing was that @cuba said the sanitizer doesnt work properly unless its eagerly loaded because the component is registered and can return data before the sanitizer is an observer of the local files data service. I eventually found that desktop is setting up this service (among others) in this `DoFinalInitForServices`: https://github.com/brave/brave-core/blob/fb013395389253be4978ab043fc230fd86ec8715/browser/profiles/brave_profile_manager.cc#L125

I didnt know about `ServiceIsCreatedWithBrowserState`, it seems like the desktop version of that (`ServiceIsCreatedWithBrowserContext`) should be used for the services started inside of that DoFinalInit method as well right?